flink-core/src/test/java/org/apache/flink/util/FileUtilsTest.java

```diff
@@ -267,6 +241,25 @@ public void testCompression() throws IOException {
 		assertDirEquals(compressDir.resolve(originalDir), extractDir.resolve(originalDir));
 	}
 
+	@Test
+	public void testCompressionOnRelativePath() throws IOException {
+
+		final java.nio.file.Path compressDir = tmp.newFolder("compressDir").toPath();
+		final java.nio.file.Path relativeCompressDir =
+			Paths.get(new File("").getAbsolutePath()).relativize(compressDir);
+		final java.nio.file.Path extractDir = tmp.newFolder("extractDir").toPath();
```

Comment:
I guess we can further extract the below common codes into `prepareTestDirectoriesForCompression -> testCompression` by providing additional two or three arguments  if you want. But i can also accept the current way.

       ```
        final java.nio.file.Path extractDir = tmp.newFolder("extractDir").toPath();
		final java.nio.file.Path originalDir = prepareTestDirectoriesForCompression(compressDir);
		final Path zip = FileUtils.compressDirectory(
			new Path(relativeCompressDir.resolve(originalDir).toString()),
			new Path(compressDir.resolve(originalDir) + ".zip"));

		FileUtils.expandDirectory(zip, new Path(extractDir.toAbsolutePath().toString()));

		assertDirEquals(relativeCompressDir.resolve(originalDir), extractDir.resolve(originalDir));
```